docs/tools.md

```diff
@@ -0,0 +1,101 @@
+
+## Development
+
+### Writing a new tool
+
+To create a tool, the `Tool` or `DynamicTool` class must be implemented from [tools base](../src/tools/base.ts). When starting to write tools, it is recommended to implement the `Tool` class. The `DynamicTool` class is an extension of the `Tool` class that allows for dynamic forms of input. Refer to the list of [examples](#examples) for some simple examples or any of the built-in tools in order to guide the creation of new tools.
+
+Tools MUST do the following:
+
+- Implement the `Tool` class:
+
+  ```typescript
+  export class MyNewTool extends Tool<MyNewToolOutput, ToolOptions, ToolRunOptions> {
+    // tool implementation
+  }
+  ```
+
+- Be given a unique name:
+
+  ```typescript
+  name = "MyNewTool";
+  ```
+
+- Provide a natural language description of what the tool does:
+
+  ❗Important: this description is used by the agent to determine when the tool should be used. It's probably the most important aspect of your tool and you should experiment with different natural language descriptions to ensure the tool is used in the correct circumstances.
+
+  ```typescript
+  description = "Takes X action when given Y input resulting in Z output";
+  ```
+
+- Declare an input schema:
+
+  This is used to define the format of the input to your tool. The agent will formalise the natural language input(s) it has received and structure them into the fields described in the tool's input. The input schema is specified using [Zod](https://github.com/colinhacks/zod).
+
+  ```typescript
+  // any Zod definition is good here, this is typical simple example
+  inputSchema = z.object({
+    // list of key-value pairs
+  });
+  ```
+
+- Implement initialisation:
+
+  The unnamed static block is executed when your tool is called for the first time. It is used for registering your tool to the agent and any other custom initialisation that may be required.
+
+  ```typescript
+  static {
+      this.register();
+  }
+  ```
+
+- Implement the `_run()` method:
+
+  ```typescript
+  protected async _run(input: ToolInput<this>, options?: BaseToolRunOptions) {
+      // insert custom code here
+      // MUST: return an instance of the output type specified in the tool class definition
+      // MAY: throw an instance of ToolError upon unrecoverable error conditions encountered by the tool
+  }
+  ```
+
+### Using tools with agents
+
+In order for a tool to be of some utility within an agent, you must enable the agent with knowledge of the tool. To do this, the tool code module must be imported into the agent and passed to the tools array during creation of a `BeeAgent`. An example can be found in the [bee agent](../examples/agents/bee.ts) or you can use a code snippet such as the one below that creates an agent with the built-in [ArXiv tool](../src/tools/arxiv.ts):
+
+```typescript
+import { ArXivTool } from "@/tools/arxiv.js";
```

Comment:
I think we should use the full package name `bee-agent-framework` instead of the `@` alias path in our docs.